shortcircuit-api/src/main/java/com/powsybl/shortcircuit/FaultResult.java

```diff
@@ -41,4 +54,26 @@ public float getThreePhaseFaultCurrent() {
         return threePhaseFaultCurrent;
     }
 
+    /**
+     * List of contributions to the three phase fault current of each connectable connected to the equipment
+     */
+    public List<FeederResult> getFeederResults() {
+        return feederResults;
+    }
+
+    public double getFeederCurrent(String feederId) {
+        try {
+            double current = 0;
+            for (FeederResult feederResult : feederResults) {
+                if (feederResult.getConnectableId().equals(feederId)) {
+                    current = feederResult.getFeederThreePhaseCurrent();
+                }
+            }
+            return current;
+        } catch (IllegalArgumentException e) {
+            LOGGER.warn("Feeder id does not exist in FeederResults");
```

Comment:
Can you be more explicit in our message with printing the id in question ?

Reply:
An exception is "exceptionnal" and should not be used to handle normal case where a data does not exist